### DISCUSSION TITLE
How to visualize pointcloud and camera poses

### DISCUSSION THREAD

**Discussion by smileyenot983** [cnr-isti-vclab/meshlab#1458] [Q&A]
Hi everyone! I am using meshlab as a convenient tool for sfm/slam visualization. I want to visualize pointcloud with camera poses. For camera poses i want them to look like frustums(attached an image). At the moment i am able to visualize the pointcloud by saving it as *.ply file and was wondering if there is any way to save corresponding camera poses into *.txt file or something else for easy visualization.
![maxresdefault](https://github.com/cnr-isti-vclab/meshlab/assets/54511486/0929db22-6123-486a-a8de-35407094f358)

**Comment by jmespadero** [comment]
Meshlab support importing the cameras from files with the Agisoft .xml or Bundler .out file formats.

You can get a plain description of the formats at https://www.agisoft.com/forum/index.php?topic=2351.msg12556#msg12556

Agisoft xml file format basically  dumps the 4x4 matrix camera as 16  values tagged into a `<transform>` tag:

` <transform>9.9979525172066075e-001 1.9951612434278700e-002 -3.3745811729752944e-003 9.2404807307462611e-001 -1.9937006769072906e-002 9.9979195775360141e-001 4.3077804390216183e-003 -3.2245823692836267e-001 3.4598262832987404e-003 -4.2396193807005631e-003 9.9998502750271023e-001 -1.0787526281678460e-002 0.0000000000000000e+000 0.0000000000000000e+000 0.0000000000000000e+000 1.0000000000000000e+000</transform>`
